src/test/java/org/broadinstitute/hellbender/engine/spark/SparkCommandLineProgramUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import org.apache.hadoop.test.MockitoMaker;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.tools.ant.taskdefs.Java;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.TestSparkProgramGroup;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class SparkCommandLineProgramUnitTest extends BaseTest {
+
+    @CommandLineProgramProperties(summary = "test", programGroup = TestSparkProgramGroup.class, oneLineSummary = "line")
+    private static class TestSparkCommandLineProgram extends SparkCommandLineProgram {
+        private static final long serialVersionUID = 1L;
+        @Override
+        protected void runPipeline(JavaSparkContext ctx) {
+        }
+    };
+
+    // Because the exact values here are system dependant in local mode, this tests are just sanity tests
+    @Test (dataProvider = "testMemoryPerCoreData")
+    public void testMemoryPerCore(JavaSparkContext context, double expected) {
+        TestSparkCommandLineProgram prog = new TestSparkCommandLineProgram();
+        prog.ctx = context;
```

Comment:
maybe call `onStartup()` rather than poking a value in there.  (that would also let you make ctx private.)